pkg/cloudprovider/vsphere/loadbalancer/access.go

```diff
@@ -0,0 +1,494 @@
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package loadbalancer
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"k8s.io/cloud-provider-vsphere/pkg/cloudprovider/vsphere/loadbalancer/config"
+)
+
+const (
+	// ScopeOwner is the owner scope
+	ScopeOwner = "owner"
+	// ScopeCluster is the cluster scope
+	ScopeCluster = "cluster"
+	// ScopeService is the service scope
+	ScopeService = "service"
+	// ScopePort is the port scope
+	ScopePort = "port"
+	// ScopeIPPoolID is the IP pool id scope
+	ScopeIPPoolID = "ippoolid"
+	// ScopeLBClass is the load balancer class scope
+	ScopeLBClass = "lbclass"
+)
+
+type access struct {
+	broker       NsxtBroker
+	config       *config.LBConfig
+	ownerTag     model.Tag
+	standardTags Tags
+}
+
+var _ NSXTAccess = &access{}
+
+// NewNSXTAccess creates a new NSXTAccess instance
+func NewNSXTAccess(broker NsxtBroker, config *config.LBConfig) (NSXTAccess, error) {
+	standardTags := Tags{
+		ScopeOwner: newTag(ScopeOwner, AppName),
+	}
+	for k, v := range config.AdditionalTags {
+		standardTags[k] = newTag(k, v)
+	}
+	return &access{
+		broker:       broker,
+		config:       config,
+		ownerTag:     standardTags[ScopeOwner],
+		standardTags: standardTags,
+	}, nil
+}
+
+func (a *access) FindIPPoolByName(poolName string) (string, error) {
+	list, err := a.broker.ListIPPools()
+	if err != nil {
+		return "", errors.Wrap(err, "listing IP pools failed")
+	}
+	for _, item := range list {
+		if item.DisplayName != nil && *item.DisplayName == poolName {
+			return *item.Id, nil
+		}
+	}
+	return "", fmt.Errorf("load balancer IP pool named %s not found", poolName)
+}
+
+func (a *access) CreateLoadBalancerService(clusterName string) (*model.LBService, error) {
+	lbService := model.LBService{
+		Description:      strptr(fmt.Sprintf("virtual server pool for cluster %s created by %s", clusterName, AppName)),
+		DisplayName:      strptr(fmt.Sprintf("cluster:%s", clusterName)),
+		Tags:             a.standardTags.Append(clusterTag(clusterName)).Normalize(),
+		Size:             strptr(a.config.LoadBalancer.Size),
+		Enabled:          boolptr(true),
+		ConnectivityPath: strptr(a.config.LoadBalancer.Tier1GatewayPath),
+	}
+	result, err := a.broker.CreateLoadBalancerService(lbService)
+	if err != nil {
+		return nil, errors.Wrapf(err, "creating load balancer service failed for cluster %s", clusterName)
+	}
+	return &result, nil
+}
+
+func (a *access) FindLoadBalancerService(clusterName string, id string) (*model.LBService, error) {
+	if id != "" {
```

Comment:
nit: this can be a bit easier to follow with an early return
```go
if id == "" {
   return a.findLoadBalancerService(a.ownerTag, clusterTag(clusterName)) 
}
...
```

Reply:
done